src/main/java/core/basesyntax/EmployeeService.java

```diff
@@ -1,10 +1,29 @@
 package core.basesyntax;
 
+import java.util.Comparator;
 import java.util.List;
 import java.util.Set;
+import java.util.TreeSet;
 
 public class EmployeeService {
     public Set<Employee> getEmployByOrder(List<Employee> employees) {
-        return null;
+        if (employees == null) {
+            throw new IllegalArgumentException("Input list cannot be null");
+        }
+        Comparator<Employee> employeeComparator = new Comparator<Employee>() {
```

Comment:
Consider using a lambda expression instead of an anonymous class for the Comparator. This will make the code more concise and easier to read.

Reply:
Lambda would simplify an expression a little bit and it would like something like that:
`Comparator<Employee> employeeComparator = (o1, o2) -> {
            int ageCompared = Integer.compare(o1.getAge(), o2.getAge());
            if (ageCompared != 0) {
                return ageCompared;
            }
            return o1.getName().compareToIgnoreCase(o2.getName());
        };`
And additionally with lambda you could later use methods from the Comparator class. It provides utility methods such as comparingInt and thenComparing which can be used to simplify the comparison logic.
       
It's fine however if you haven't been through this material yet :) you can leave it as it is. Just remove the Employee type on the right side of the Comparator expression, it's redundant.